shared/src/commonMain/kotlin/com/automattic/myapplication/shared/internal/EventScheduler.kt

```diff
@@ -15,8 +15,10 @@ internal class EventScheduler(
 
     init {
         coroutineScope.launch {
-            localRepository.observe().collect { event ->
-                remoteRepository.send(event)
+            localRepository.observe().collect { events ->
+                if (events.isNotEmpty()) {
+                    remoteRepository.send(events.first())
```

Comment:
Why just the first? Shouldn't we iterate over all events?

Reply:
The idea I was having was to pick the first (or e.g. "the most important") event, process it, send and after successful sending - remove it from DB which will trigger the `observe` again. 

But we could e.g. order events here from "the most important" to the less and then send whole list to `RemoteRepository` - both works fine for me.

---
"most important" - TBD, maybe number of failed attempts or something like this